File: tests/fixtures/sample_project_csharp/src/Auth/Handlers.cs
using Acme.Db;
using Acme.Users;

namespace Acme.Auth;

/// <summary>Authentication request handlers — login and session verification.</summary>
public class Handlers
{
    /// <summary>Validate credentials and issue an access token.</summary>
    public static string AuthenticateUser(Connection conn, string email)
    {
        User user = Crud.GetUserByEmail(conn, email);
        return JwtUtils.EncodeToken(user.Email);
    }

    /// <summary>Return true if the access token is currently valid.</summary>
    public static bool VerifySession(string token)
    {
        return JwtUtils.DecodeToken(token) != null;
    }
}

File: tests/fixtures/sample_project_php/src/Auth/Handlers.php
<?php

namespace Acme\Auth;

use Acme\Db\Connection;
use Acme\Users\Crud;

/** Authentication request handlers — login and session verification. */
class Handlers
{
    /** Validate credentials and issue an access token. */
    public static function authenticateUser($conn, $email)
    {
        $user = Crud::getUserByEmail($conn, $email);
        return JwtUtils::encodeToken($user->email);
    }

    /** Return true if the access token is currently valid. */
    public static function verifySession($token)
    {
        return JwtUtils::decodeToken($token) !== null;
    }
}
